baseservice/service_addons/loop_health_addon.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import threading
+from time import time
+
+from baseservice.base_service import ServiceState
+from baseservice.server_loop_service import ServerLoopService, LoopMetrics
+
+
+class LoopHealthAddon:
+    def __init__(self, service: ServerLoopService, *,
+                 stop_after_consecutive_failures: int = -1,
```

Comment:
:nit: The arguments here sound like boolean typed although you meant they'll represent a numeric value. 
consider renaming them. 
suggestion: `max_consecutive_failures` and `inactivity_timeout`